### PR TITLE
Add test for extending a constructor with null .prototype

### PR DESCRIPTION
**Pull request by bakkot**
This tests for the behavior being proposed in https://github.com/tc39/ecma262/pull/755.

Don't merge until we have consensus to merge that PR.

### CODE REVIEW

File: test/language/statements/class/constructor-inferred-observable-iteration.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ description: >
 info: |
   [...]
   10. If constructor is empty, then
-      a. If ClassHeritageopt is present and protoParent is not null, then
+      a. If ClassHeritageopt is present and superclass is not null, then
```
Comment (ljharb) [inline test/language/statements/class/constructor-inferred-observable-iteration.js]: `Classheritageopt` seems missing an underscore

File: test/language/statements/class/subclass/class-definition-null-proto-missing-return-override.js [hunk 2]
```diff
@@ -16,11 +16,9 @@ info: |
      [...]
   6. Else,
      [...]
-     e. If superclass is null, then
-        i. Let protoParent be null.
-        ii. Let constructorParent be the intrinsic object %FunctionPrototype%.
+     b. Let superclass be the result of evaluating ClassHeritage.
   [...]
-  15. If ClassHeritageopt is present and protoParent is not null, then set F's
+  15. If ClassHeritageopt is present and superclass is not null, then set F's
```
Comment (ljharb) [inline test/language/statements/class/subclass/class-definition-null-proto-missing-return-override.js]: also here

File: test/language/statements/class/subclass/class-definition-null-proto-super.js [hunk 3]
```diff
@@ -13,11 +13,9 @@ info: |
      [...]
   6. Else,
      [...]
-     e. If superclass is null, then
-        i. Let protoParent be null.
-        ii. Let constructorParent be the intrinsic object %FunctionPrototype%.
+     b. Let superclass be the result of evaluating ClassHeritage.
   [...]
-  15. If ClassHeritageopt is present and protoParent is not null, then set F's
+  15. If ClassHeritageopt is present and superclass is not null, then set F's
```
Comment (ljharb) [inline test/language/statements/class/subclass/class-definition-null-proto-super.js]: and here

File: test/language/statements/class/subclass/class-definition-null-proto-this.js [hunk 4]
```diff
@@ -15,11 +15,9 @@ info: |
      [...]
   6. Else,
      [...]
-     e. If superclass is null, then
-        i. Let protoParent be null.
-        ii. Let constructorParent be the intrinsic object %FunctionPrototype%.
+     b. Let superclass be the result of evaluating ClassHeritage.
   [...]
-  15. If ClassHeritageopt is present and protoParent is not null, then set F's
+  15. If ClassHeritageopt is present and superclass is not null, then set F's
```
Comment (ljharb) [inline test/language/statements/class/subclass/class-definition-null-proto-this.js]: and here

File: test/language/statements/class/subclass/class-definition-null-proto.js [hunk 5]
```diff
@@ -16,15 +16,13 @@ info: |
...
-        i. Let protoParent be null.
-        ii. Let constructorParent be the intrinsic object %FunctionPrototype%.
+     b. Let superclass be the result of evaluating ClassHeritage.
      [...]
   7. Let proto be ObjectCreate(protoParent).
   8. If ClassBodyopt is not present, let constructor be empty.
   9. Else, let constructor be ConstructorMethod of ClassBody.
   10. If constructor is empty, then
-      a. If ClassHeritageopt is present and protoParent is not null, then
+      a. If ClassHeritageopt is present and superclass is not null, then
```
Comment (ljharb) [inline test/language/statements/class/subclass/class-definition-null-proto.js]: here too

**Comment by bakkot** [comment]
This is now ok to merge (assuming it looks good).

**Comment by leobalter** [comment]
any missing push?

**Comment by tcare** [comment]
Awaiting @ljharb signoff

**Comment by ljharb** [comment]
(It appears the latest changes haven't yet been pushed up)

**Comment by bakkot** [comment]
Sorry for the confusion - I didn't have any more changes; I just meant that [the accompanying PR](https://github.com/tc39/ecma262/pull/755) to ecma262 has been merged, and so this is now an accurate test.

@ljharb, convention in the spec-text in headers in this repo is to omit that underscore. That's how it's written everywhere else. I agree it looks odd, but don't want to mess with it in this PR :)

**Comment by ljharb** [comment]
@bakkot ah ok, super weird but agreed on being out of scope

**Pull request merged**